### ISSUE TITLE
Seq2Seq While opperation conversion issue

### ISSUE THREAD

**Issue by rmarrugat** [onnx/tensorflow-onnx#1687]
**Describe the bug**
We have a Seq2Seq model built with tensorflow_addons, which we converted to protobuf and ran inference with it using the tensorflow C API. Now we want to convert the model into onnx.

The model contains some HashTableV2 to convert from String to int64 and the other way around, loop-like functions (While operations) and even TensorList operations.

I managed to complete the conversion from my protobufed model into an onnx one, but the resulting model is erroneous as, for example, it still contains TensorList operations not supported by onnx with UNDEFINED datatype. As far as I understand, these operations are generally removed as they would be handled by "While onnx node", as seen in the While class conversion of _controlflow.py_. The problem is that the _tf_optimize_ function of the _tf_loader_ optimized my graph by "unwrapping" the While operations of my initial graph into Loop operations, so the aforementioned While conversion is never called and the TensorList operations are never handled.

If I try to disable these initial optimizations to keep my While operations, a problem emerges with the conversions of the HashTableV2 as calls contained in inner graphs (functions) are not able to find the table_node specified in outter graphs.

I hope my explanation was somewhat understandable.

**System information**

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Tensorflow Version: 2.5
- Python version: 3.7

**To Reproduce**
I attach the txt version of my protobufed model for clarity.
[model.pb.txt](https://github.com/onnx/tensorflow-onnx/files/7066430/AEAT_tfpb2.pb.txt)

**Comment by TomWildenhain-Microsoft** [comment]
Hi @rmarrugat, when I tried to load your model into tensorflow I got 
```
strides[0] must be non-zero for '{{node strided_slice}} = StridedSlice[Index=DT_INT32, T=DT_INT64, _output_shapes=[[?,?]], begin_mask=3, ellipsis_mask=0, end_mask=1, new_axis_mask=0, shrink_axis_mask=0](PadV2, strided_slice/stack, strided_slice/stack_1, strided_slice/stack_2)' with input shapes: [?,?], [2], [2], [2] and with computed input tensors: input[3] = <0 0>.
```
Also I had to modify the model.pb.txt a bit to parse.  Looks like a bug in protobuf.  
```
  attr {
    key: "_"read"_only_resource_inputs"
    value {
      list {
      }
    }
  }
```

Can you please upload your original saved_model or model.pb with the command you used to convert it?

**Comment by rmarrugat** [comment]
Yes, I am using `tensorflow.python.framework.convert_to_constants.convert_variables_to_constants_v2` already.
I did not try `tf2onnx.convert.from_keras`, but I did try `tf2onnx.convert.from_saved_model` with no success.

Looking at the `.pb.txt` the shapes looked alright to me, but `tensorflow.python.framework.convert_to_constants.convert_variables_to_constants_v2` already deals with this as it retrieves the `tensor_data` from the session to save it into the graph as `Const` and does compute the shapes to save it as a `tensor_proto`.

I don't have a python script that runs the model, but a `Rust` one. I could try to develop a minimal script that runs the model on python.
